R/epi_calendar.R

```diff
@@ -23,34 +23,32 @@
 #'
 #' @examples
 #' # Check if a date is within the epidemiological season
-#' epi_calendar(as.Date("2023-09-15"), start = 40, end = 20)
-#' # Expected output: "2022/2023"
+#' epi_calendar(as.Date("2023-09-15"), start = 21, end = 20)
+#' # Expected output: "2023/2024"
 #'
-#' epi_calendar(as.Date("2023-05-01"), start = 40, end = 20)
+#' epi_calendar(as.Date("2023-05-30"), start = 40, end = 20)
 #' # Expected output: "out_of_season"
 #'
-#' epi_calendar(as.Date("2023-01-15"), start = 40, end = 20)
+#' epi_calendar(as.Date("2023-01-15"), start = 21, end = 20)
 #' # Expected output: "2022/2023"
 #'
-#' epi_calendar(as.Date("2023-12-01"), start = 40, end = 20)
+#' epi_calendar(as.Date("2023-12-01"), start = 21, end = 20)
 #' # Expected output: "2023/2024"
-epi_calendar <- Vectorize(function(date, start = 40, end = 20) {
+epi_calendar <- Vectorize(function(date, start = 21, end = 20) {
   # Compute the current week
-  current_week <- as.integer(format(x = date, "%U"))
+  current_week <- as.integer(format(x = date, "%V"))
 
-  if (current_week <= start & end <= current_week) {
+  if (!(current_week >= start | current_week <= end)) {
```

Comment:
Is this logic sufficient?

If the start and end are within a year, say start = 20 and end = 40, then 
out of season would be "current_week < start || current_week > end"

If they are not ordered, as you examples, start = 40 and end = 20, then 
out of season would be "current_week > end && current_week < start"